neo/Network/LocalNode.cs

```diff
@@ -177,22 +213,12 @@ public static void AllowHashes(IEnumerable<UInt256> hashes)
 
         private void Blockchain_PersistCompleted(object sender, Block block)
         {
-            Transaction[] remain;
             lock (mem_pool)
             {
                 foreach (Transaction tx in block.Transactions)
-                {
                     mem_pool.Remove(tx.Hash);
-                }
-                if (mem_pool.Count == 0) return;
-
-                remain = mem_pool.Values.ToArray();
-                mem_pool.Clear();
```

Comment:
We need to ensure that all transactions in the `mem_pool` are verified.

Reply:
Ok, I’ll change it to put the ones that aren’t verified back in the tmp_pool and update the pull request.

Reply:
Made the changes to ensure all transactions in the mem_pool are always verified.